pandas/tseries/tools.py

```diff
@@ -510,6 +510,9 @@ def coerce(values):
         # we allow coercion to if errors allows
         return to_numeric(values, errors=errors)
 
+    # prevent overflow in case of int8 or int16
+    arg = arg.astype('int64', copy=False)
```

Comment:
@jreback What should be the output in case of string type ? Or should it raise ValueError?
```python
df = pd.DataFrame({'year': [2000, 2001], 'month': [1, 1], 'day': [1, 1]})
pd.to_datetime(df.astype(str))
```